test/integ/test_cursor.py

```diff
@@ -644,109 +644,96 @@ def test_executemany(conn, db_parameters):
     Notes:
         The binding data type is dict and tuple, respectively.
     """
+    table_name = random_string(5, "test_executemany_")
     with conn() as cnx:
-        c = cnx.cursor()
-        fmt = "insert into {name}(aa) values(%(value)s)".format(
-            name=db_parameters["name"]
-        )
-        c.executemany(
-            fmt,
-            [
-                {"value": "1234"},
-                {"value": "234"},
-                {"value": "34"},
-                {"value": "4"},
-            ],
-        )
-        cnt = 0
-        for rec in c:
-            cnt += int(rec[0])
-        assert cnt == 4, "number of records"
-        assert c.rowcount == 4, "wrong number of records were inserted"
-        c.close()
+        with cnx.cursor() as c:
+            c.execute(f"create temp table {table_name} (aa number)")
+            c.executemany(
+                f"insert into {table_name}(aa) values(%(value)s)",
+                [
+                    {"value": 1234},
+                    {"value": 234},
+                    {"value": 34},
+                    {"value": 4},
+                ],
+            )
+            assert c.fetchone()[0] == 4, "number of records"
+            assert c.rowcount == 4, "wrong number of records were inserted"
 
-        c = cnx.cursor()
-        fmt = "insert into {name}(aa) values(%s)".format(name=db_parameters["name"])
-        c.executemany(
-            fmt,
-            [
-                (12345,),
-                (1234,),
-                (234,),
-                (34,),
-                (4,),
-            ],
-        )
-        rec = c.fetchone()
-        assert rec[0] == 5, "number of records"
-        assert c.rowcount == 5, "wrong number of records were inserted"
-        c.close()
+        with cnx.cursor() as c:
+            fmt = "insert into {name}(aa) values(%s)".format(name=db_parameters["name"])
```

Comment:
should we also use the temp table here?

Reply:
It's not necessary temporary objects are cleaned up when the `SnowflakeConnection` gets closed, cursors don't affect it